protected/controllers/CsvController.php

```diff
@@ -210,6 +209,210 @@ private function createNewPatient($patient)
         if(!$new_patient->save()){
             return $new_patient->getErrors();
         }
+        //patient contact assignments
+
+        //referred to
+        if(isset($patient['referred_to_first_name']) and isset($patient['referred_to_last_name'])){
+            $referred_to = null;
+            //Find if exists
+            $referred_to = User::model()->findByAttributes(array(
+                'first_name' => $patient['referred_to_first_name'],
+                'last_name' => $patient['reffered_to_last_name'],
+            ));
+            if($referred_to === null){
+                $errors[] = 'Cannot find referred to user';
```

Comment:
You might want to give a bit more information here, like 'Cannot find the referred to user of "$first_name $last_name" for patient "$patient->getName()"'